src/FsPickler/PicklerGeneration/PicklerResolution.fs

```diff
@@ -33,19 +33,33 @@ let resolvePickler (resolver : IPicklerResolver)
         // step 2: create an uninitialized pickler instance and register to the local cache
         let p0 = PicklerGenerator.CreateUninitialized shape
         mkEarlyBinding p0
+        
+        // get base types and implemented interfaces
+        let baseAndInterfaceTypes = [
+            let rec getBaseTypes (t : Type) = seq {
+                if t.BaseType <> null then
+                    yield t.BaseType
+                    yield! getBaseTypes t.BaseType
+            }
+            yield! getBaseTypes t
+            yield! t.GetInterfaces()

```

Comment:
NB interfaces can also have base types

Reply:
As far as I know, ``GetInterfaces`` flattens the hierarchie. Simple test:

```F#
type I = interface end
type I2 = interface end

type I3 =
    inherit I
    inherit I2
	
Type A() =
    interface I
    interface I2
	
type B() =
    inherit A()
    interface I3
```

```F#
(new B()).GetType().GetInterfaces() |> Array.Map(fun t -> t.Name)

> val it : string [] = [|"I2"; "I"; "I3"|]
```